src/GT.Core/Services/Impl/GTCompanyService.cs

```diff
@@ -92,14 +153,114 @@ public async Task<bool> ExistsByNameAsync(string name)
 			}
 		}
 
-		public Task<CompanyDTO> GetAsync()
+		public async Task<List<CompanyDTO>> GetAsync()
 		{
-			throw new NotImplementedException();
+			try
+			{
+				var entities = await _companyRepository.GetAll().ToListAsync();
+				var companyDTOs = new List<CompanyDTO>();
+
+				foreach (var entity in entities)
+				{
+					// TODO add automapper
+					companyDTOs.Add(new CompanyDTO
+					{
+						Id = entity.Id,
+						Name = entity.Name,
+						CompanyLogoId = entity.CompanyLogoId,
+					});
+				}
+
+				return companyDTOs;
+			}
+			catch (Exception e)
+			{
+				_logger.LogError(e.Message);
+				return null;
+			}
 		}
 
-		public Task<CompanyDTO> GetByIdAsync(string companyId)
+		public async Task<CompanyDTO> GetByIdAsync(string id)
 		{
-			throw new NotImplementedException();
+			if (id is null)
+			{
+				_logger.LogWarning($"Attempted to get entity with null reference id argument. {nameof(GetByIdAsync)}");
+				return null;
+			}
+
+			try
+			{
+				// Get entity
+				var entity = await _companyRepository
+					.GetAll()
+					.FirstOrDefaultAsync(e => e.Id == id);
+
+				if (entity == null)
+				{
+					_logger.LogInformation($"Entity with id {id} not found.");
+					return null;
+				}
+
+				// Map entity to DTO
+				var company = new CompanyDTO()
+				{
+					Id = entity.Id,
+					Name = entity.Name,
+					CompanyLogoId = entity.CompanyLogoId,
+				};
+
+				return company;
+			}
+			catch (Exception e)
+			{
+				_logger.LogError(e.Message);
+				return null;
+			}
+		}
+
+		public async Task<bool> UpdateAsync(CompanyDTO companyDTO, string name)
+		{
+			try
+			{
+				if (companyDTO.Id is not null && name is not null)
+				{
+					if (await ExistsByNameAsync(name))
+					{
+						var entityToUpdate = await _companyRepository.GetAll().Where(e => e.Id == companyDTO.Id).FirstOrDefaultAsync();
```

Comment:
Lambda expression can be wrapped in the .FirstOrDefaultAsync() call, eliminating the need for the .Where() clause.

Reply:
Fixed